full-service/src/service/t3_sync.rs

```diff
@@ -0,0 +1,152 @@
+            T3Connection::new(&config.t3_uri, config.t3_api_key.clone(), logger.clone());
+
+        let join_handle = Some(
+            thread::Builder::new()
+                .name("t3_sync".to_string())
+                .spawn(move || {
+                    log::debug!(logger, "T3 Sync thread started.");
+
+                    let conn = &mut wallet_db
+                        .get_pooled_conn()
+                        .expect("failed getting wallet db connection");
+
+                    loop {
+                        if thread_stop_requested.load(Ordering::SeqCst) {
+                            log::debug!(logger, "T3SyncThread stop requested.");
+                            break;
+                        }
+
+                        match sync_txos(conn, &t3_connection, &logger) {
+                            Ok(()) => (),
+                            Err(e) => log::error!(&logger, "Error during t3 sync:\n{:?}", e),
+                        }
+
+                        // This sleep is to allow other API calls that need access to the database a
+                        // chance to execute, because the t3 sync process requires a write lock on
+                        // the database.
+                        thread::sleep(std::time::Duration::from_millis(T3_SYNC_INTERVAL));
+                    }
+                    log::debug!(logger, "T3SyncThread stopped.");
+                })
+                .expect("failed starting t3 sync thread"),
+        );
+
+        Self {
+            join_handle,
+            stop_requested,
+        }
+    }
+
+    pub fn stop(&mut self) {
+        self.stop_requested.store(true, Ordering::SeqCst);
+        if let Some(join_handle) = self.join_handle.take() {
+            join_handle.join().expect("T3SyncThread join failed");
+        }
+    }
+}
+
+impl Drop for T3SyncThread {
+    fn drop(&mut self) {
+        self.stop();
+    }
+}
+
+pub fn sync_txos(
+    conn: Conn,
+    t3_connection: &T3Connection,
+    logger: &Logger,
+) -> Result<(), T3SyncError> {
+    // get all txos from the database that haven't synced yet to t3 and that have an
+    // authenticated sender memo
+    log::debug!(logger, "Syncing txos to t3");
+
+    let txos = Txo::get_txos_that_need_to_be_synced_to_t3(Some(TXO_CHUNK_SIZE), conn)?;
+
+    for txo in txos {
+        let txo_memo = txo.memo(conn)?;
+        let memo = match txo_memo {
+            TxoMemo::AuthenticatedSender(memo) => memo,
+            _ => continue,
+        };
+
+        sync_txo(&txo, &memo, t3_connection)?;
+        txo.update_is_synced_to_t3(true, conn)?;
+    }
+
+    Ok(())
+}
+
+fn sync_txo(
+    txo: &Txo,
+    memo: &AuthenticatedSenderMemo,
+    t3_connection: &T3Connection,
+) -> Result<(), T3SyncError> {
+    let mut transparent_transaction = TransparentTransaction::new();
+
+    let sender_address_hash_bytes = hex::decode(&memo.sender_address_hash)?;
+    transparent_transaction.set_sender_address_hash(sender_address_hash_bytes);
```

Comment:
I don't see where we are setting the recipient address hash.

Reply:
That's a todo! Thanks 😊

Reply:
Consider adding an explicit TODO
```suggestion
    transparent_transaction.set_sender_address_hash(sender_address_hash_bytes);
    // TODO: Set recipient_address_hash
```

Reply:
WIll do that in the future, bad habit that I don't! Thanks :)